src/Workspaces/Core/Portable/Workspace/WorkspaceChangeEventArgs.cs

```diff
@@ -4,6 +4,15 @@
 
 namespace Microsoft.CodeAnalysis
 {
+    /// <summary>
+    /// The <see cref="EventArgs"/> describing any kind of workspace change.
+    /// </summary>
+    /// <remarks>
+    /// When linked files are edited, one document change event is fired per linked file. All of
+    /// these events contain the same OldSolution, and they all contain the same NewSolution. This
+    /// is so that we can trigger document change events on all affected documents without
+    /// reporting intermediate states in which the linked file contents do not match.
+    /// </remarks>
     public class WorkspaceChangeEventArgs : EventArgs
```

Comment:
I'll add the comment for the linked file case, but if a refactoring touches multiple files, it's applied using TryApplyChanges, which changes one document at a time.